runtime/http.go

```diff
@@ -49,47 +59,72 @@ func (*httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			http.Error(w, "failed ReadAll in PUT", http.StatusBadRequest)
 			return
 		}
-		unlock, eerr := enableAndLock(key, string(v))
-		if eerr != nil {
-			http.Error(w, "failed to set failpoint "+string(key), http.StatusBadRequest)
-			return
+
+		fpMap := map[string]string{key: string(v)}
+		if strings.EqualFold(key, "failpoints") {
+			fpMap, err = parseFailpoints(string(v))
+			if err != nil {
+				http.Error(w, fmt.Sprintf("fail to parse failpoint: %v", err), http.StatusBadRequest)
+				return
+			}
 		}
-		w.WriteHeader(http.StatusNoContent)
-		if f, ok := w.(http.Flusher); ok {
-			// flush before unlocking so a panic failpoint won't
-			// take down the http server before it sends the response
-			f.Flush()
+
+		for k, v := range fpMap {
+			if err := enable(k, v); err != nil {
+				http.Error(w, fmt.Sprintf("fail to set failpoint: %v", err), http.StatusBadRequest)
+				return
+			}
 		}
-		unlock()
+		writeCodeAndFlush(w, http.StatusNoContent)
```

Comment:
Could you just defer `w.Flush()` in the `ServeHTTP` function? Defer has a deterministic order LIFO. 

Below code guarantees that `flush` will be called before `Unlock` as it should be cleaner.
```
func (*httpHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	failpointsMu.Lock()
	defer failpointsMu.Unlock()
	defer flush(w)
```

Reply:
makes sense & updated.